src/machines/modelingMachine.ts

```diff
@@ -2310,34 +2310,174 @@ export const modelingMachine = setup({
         }
 
         // Extract inputs
-        const ast = kclManager.ast
+        let ast = kclManager.ast
+        let modifiedAst = structuredClone(ast)
+        let focusPath: PathToNode[] = []
         const { nodeToEdit, selection, radius } = input
 
-        // If this is an edit flow, first we're going to remove the old node
-        if (nodeToEdit) {
-          const oldNodeDeletion = deleteNodeInExtrudePipe(nodeToEdit, ast)
-          if (err(oldNodeDeletion)) return oldNodeDeletion
-        }
-
         const parameters: FilletParameters = {
           type: EdgeTreatmentType.Fillet,
           radius,
         }
+
         const dependencies = {
           kclManager,
           engineCommandManager,
           editorManager,
           codeManager,
         }
 
-        // Apply fillet to selection
-        const filletResult = await applyEdgeTreatmentToSelection(
-          ast,
-          selection,
-          parameters,
-          dependencies
+        // Apply or edit fillet
+        if (nodeToEdit) {
+          // Edit existing fillet
+          // selection is not the edge treatment itself,
+          // but just the first edge in the fillet expression >
+          // we need to find the edgeCut artifact
+          // and build a new selection from it
+          // TODO: this is a bit of a hack, we should be able
+          // to get the edgeCut artifact from the selection
+          const firstSelection = selection.graphSelections[0]
+          const edgeCutArtifact = Array.from(
+            kclManager.artifactGraph.values()
+          ).find(
+            (artifact) =>
+              artifact.type === 'edgeCut' &&
+              artifact.consumedEdgeId === firstSelection.artifact?.id
+          )
+          if (!edgeCutArtifact || edgeCutArtifact.type !== 'edgeCut') {
+            return new Error(
+              'Failed to retrieve edgeCut artifact from sweepEdge selection'
+            )
```

Comment:
If we want this to throw as toast we need return Promise.reject (sorry this sugestion might mess with linting)


```suggestion
            return Promise.reject(
              new Error('Failed to retrieve edgeCut artifact from sweepEdge selection')
            )
```